docs: Blog "Pinecone Provider Now Available for Pulumi" (#3819)
pulumi/pulumi-hugo: themes/default/content/blog/pinecone-serverless/index.md

 +   pulumi:pulumi:Stack              pinecone-ts-dev    create
 +   └─ pinecone:index:PineconeIndex  my-pinecone-index  create

Outputs:
    output: {
        value: output<string>
    }

Resources:
    + 2 to create

Do you want to perform this update? yes
Updating (dev)

     Type                             Name               Status
 +   pulumi:pulumi:Stack              pinecone-ts-dev    created (6s)
 +   └─ pinecone:index:PineconeIndex  my-pinecone-index  created (3s)

Outputs:
    output: {
        value: "my-pinecone-index-t2khngi.svc.apw5-4e34-81fa.pinecone.io"
    }

Resources:
    + 2 created

Duration: 8s
```

Have a look into your Pinecone console, you should see newly created serverless index appear and ready to use.

![Pinecone Console](./console.png)

### Deleting a Pinecone Index

Deleting a Pinecone index is as simple as creating one. You can delete a Pinecone Serverless index by simply running following command:

```shell
pulumi destroy
```

This will delete the Pinecone index for you.

{{% notes type="warning" %}}
Be aware that this will also delete all the data stored in the index.
{{% /notes %}}

### Programmatically Creating a Pulumi Pinecone Provider

> There are situations where you might want to create a Pinecone index programmatically as part of multiple providers in one Pulumi program. Examples of this could be the creation additional infrastructure in AWS or a GitHub repository to hold your application code.

Reviewer: ```suggestion
By default, Pulumi will use the [default provider configuration](/docs/concepts/resources/providers/#default-provider-configuration) for the Pinecone provider. However, there may be situations where you need to use a different configuration. In that case, programmatically creating an explicit Pinecone provider with its own configuration is how to meet that requirement. You might also need to create one or more explicit Pinecone providers if you needed to deploy indexes in multiple Pinecone accounts, for example.
```

> You can do this by creating a Pinecone Provider programmatically and passing the provider in the resource creation. Here is an example code on how this could look like.

Reviewer: ```suggestion
Here is an example on how to programmatically create a Pinecone provider and then pass the provider when creating an index.
```

```typescript
const myPinceconProvider = new pinecone.Provider("my-pinecone-provider", {
    APIKey: "my-pinecone-api-key",
});

const myPineconeIndex = new pinecone.PineconeIndex("my-pinecone-index", {
    name: "my-pinecone-index",
    dimension: 512,
    metric: pinecone.IndexMetric.Cosine,
    spec: {
        serverless: {
            cloud: pinecone.ServerlessSpecCloud.Aws,
            region: "us-west-2",
        },
    },
}, {

>     provider: myPinceconProvider,

Reviewer: ```suggestion
    provider: myPineconeProvider,
```

});
```

## Conclusion 💡

> As shown in this blog post, the Pinecone Provider for Pulumi allows you to seamlessly integrate and manage your Pinecone  using any Pulumi-supported language. This enables platform engineers and developers to maintain easily manageable and reproducible infrastructure as code for their Pinecone indexes.

Reviewer: ```suggestion
As shown in this blog post, the Pinecone Provider for Pulumi allows you to seamlessly integrate and manage your Pinecone indexes---including the new serverless indexes---using any Pulumi-supported language. This enables platform engineers and developers to maintain easily manageable and reproducible infrastructure as code for their AI applications that use Pinecone indexes.
```
